nixos/lib/testing/name.nix

```
@@ -0,0 +1,7 @@
+{ lib, ... }:
+{
+  options.name = lib.mkOption {
+    description = "The name of the test.";
```

Was just scrolling through it to pass time and see the matrix comments, spotted this:
```suggestion
    description = "The name of the test";
```

---

Though then again, maybe nixpkgs has gotten big enough that the parts I touch have a slightly different style than the parts you touch. Who knows.

---

If there's fewer than one sentence, it doesn't need a period. That makes sense, but the real problem is that I didn't write documentation for this option. The only info I added was "this option is part of the test somehow".

It's basically filler, like those cringy generated accessor docs you may find in some Java code.

---

I mean I hope people didn't write that stuff :thinking:

---

New doc:

> The name of the test.
>
> This is used in the derivation names of the [`driver`](#opt-driver) and [`test`](#opt-test) runner.

Seems like I've kept the period. Might make sense, depending on whether it's more of a sentence or more of a title. I think it's neither :shrug:.
I'll happily change it either way if it's the only thing this merge will depend on. I think we ought to fry bigger fish though.